File: my-folder/0014-longest-common-prefix/solution.c
char* longestCommonPrefix(char** strs, int strsSize) {
    char *s = strs[0];
    
    for (size_t i = 0; s[i] != 0; i++) {
        for (size_t j = 1; j < strsSize; j++) {
            if (s[i] != strs[j][i]) {
                s[i] = 0;
                goto exit;
            }
        }
    }
exit:
    return s;
}

File: my-folder/0014-longest-common-prefix/solution.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
    let mut prefix = &strs[0][..];

    for s in &strs[1..] {
        while !s.starts_with(prefix) {
            if prefix.is_empty() {
                return "".to_string();
            }
            prefix = &prefix[..prefix.len() - 1];
        }
    }

    prefix.to_string()
    }
}
